src/RulesTest/Extensions.cs

```diff
@@ -23,5 +25,17 @@ public static IEnumerable<int> Difference(this IEnumerable<int> allItems, IEnume
                    where !itemsToRemove.Contains(item)
                    select item;
         }
+
+        public static string[] GetFilteredFrameworkIds(string[] exclude, string[] additions)
```

Comment:
I like the idea of a utility method - personally, I think client code is easier to follow if this method just returns the raw list of known frameworks. Then there are no parameters and the caller is in charge of excluding or appending additional IDs. But both approaches are good with me!

Reply:
Sounds reasonable to me, I agree that's more readable so I've switched to that approach :)